Octokit/Clients/TeamsClient.cs

```diff
@@ -368,15 +425,17 @@ public async Task<bool> AddRepository(int id, string organization, string repoNa
         }
 
         /// <summary>
-        /// Add a repository to the team
+        /// Add or update team repository permissions (Legacy)
+        /// Deprecation Notice: This endpoint route is deprecated and will be removed from the Teams API.
+        /// We recommend migrating your existing code to use the new "Add or update team repository permissions" endpoint.
         /// </summary>
         /// <param name="id">The team identifier.</param>
         /// <param name="organization">Org to associate the repo with.</param>
         /// <param name="repoName">Name of the repo.</param>
         /// <param name="permission">The permission to grant the team on this repository.</param>
         /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
         /// <returns></returns>
-        [ManualRoute("PUT", "/orgs/{org}/team/{team_slug}/repos/{owner}/{repo}")]
+        [ManualRoute("PUT", "/teams/{team_id}/repos/{owner}/{repo}")]
```

Comment:
What's the reason you went with the strategy of changing routes here and adding new ones with a different method name to the client? 

I'm naively thinking it might be easier to avoid a breaking change here by isolating the new functionality to new methods.

Reply:
This is the legacy endpoint that is currently implemented in the project, however, it gets the endpoint uri from ApiUrls.TeamRepository which returns the /teams/{team_id}/repos/{owner}/{repo}, so I've decided to update the ManualRoute attribute value to match the correct url from the docs.